### PR TITLE
Richmoe notification refactor

### PR DESCRIPTION
**Pull request by RichMoeMSFT**
Major refactor of the Node sample
- splitting notification to a separate project;
- refactor readmes and apply new sample standards
- refactored to be fully functional from local builds (e.g. ngrok)

Notes: 
- auth functionality not refactored/tested
- compose extension sample not refactored/tested

### CODE REVIEW

File: Node/SampleApp/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            "type": "node",
+            "request": "launch",
+            "name": "Launch Program",
+            "program": "${workspaceRoot}\\app.js",            
+            "env": {
+                "NODE_ENV": "development",
+                "TEAMS_APP_ID": "", //This is the Teams App ID from your Manifest
+                "MICROSOFT_APP_ID": "", //Bot ID from Bot Framework
+                "MICROSOFT_APP_PASSWORD": "", //Bot password from Bot Framework
+                "CONNECTOR_APP_ID": "", //Registered Connector ID
```
Comment (jotrick) [inline Node/SampleApp/.vscode/launch.json]: Do you not need a password/secret for the connect?

File: Node/SampleApp/.vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            }
+        },
+        {
+            "type": "node",
+            "request": "launch",
+            "name": "Launch Teams Debug",
+            "program": "${workspaceRoot}\\app.js",
+            "env": {
+                "NODE_ENV": "development",
+                "TEAMS_APP_ID": "[Your Teams App GUID Here]", //This is the Teams App ID from your Manifest
```
Comment (jotrick) [inline Node/SampleApp/.vscode/launch.json]: Why are these env. variables not treated like they are above? (just leaving them blank)

File: Node/SampleApp/README.md [hunk 3]
```diff
@@ -1,53 +1,60 @@
...
-### Connectors
-You can setup the connector through the add a connector dialog, or you can set up a webhook and use the URL below to trigger messages
-
-To trigger a message just paste this url in any browser:
-https://teamsnodesample.azurewebsites.net/api/messages/connector/send?webhook_url=[webhook url]
+## Code Highlights
 
-The triggered message gives you buttons to send yourself more messages…this should be fun.
+### Tabs
+The Sample app demonstrates simple Configurable Tabs in channels and Static Tabs in personal view.  Add "auth=1" as a query parameter to tab URLs to turn on the authentication flow, which signs you in to Azure AD and gets your display name from Microsoft Graph.  Note that we have provided an alternate manifest.json with this auth flag set.
```
Comment (jotrick) [inline Node/SampleApp/README.md]: Wouldn't mind knowing more about this - as in what urls in which files - something like that.  I just don't know what this means right now.

File: Node/SampleApp/README.md [hunk 4]
```diff
@@ -1,53 +1,60 @@
...
+To register a new Connector:
+1) You'll need to register a new connector in the Connector Developer Portal, Follow the steps here: [Registering your connector](https://msdn.microsoft.com/en-us/microsoft-teams/connectors#registering-your-connector)
+2) Ensure you have both Teams and Groups checkboxes selected.
+3) For the Landing page for groups during registration, you'll use our sample code's setup endpoint: `https://[BASE_URI]/connector/setup`
+4) For the Redirect URL during registration, you'll use our sample code's registration endpoint:  `https://[BASE_URI]/api/message/connector/register`
+* In both steps 3 & 4, `[BASE_URI]` is the full URI for your running sample.
+5) In the manifest.json file, update: `connectors.connectorId` to use your new Connector ID, which you can retrieve via the Connector Developer Portal's Copy Code or Download Manifest buttons.
+6) In your launch.json / debug configuration, set the `CONNECTOR_APP_ID` environment variable to be your new Connector ID.
+
+To simply illustrate the Connector functionality, you can instead utlize the built-in Incoming Webhook connector:
```
Comment (jotrick) [inline Node/SampleApp/README.md]: Wouldn't mind a tiny more about the fact that this is the way to test it without needing to do the registration process above.

File: Node/SampleApp/graph/httpsrequesthelper.js [hunk 5]
```diff
@@ -16,7 +16,7 @@ var authHelper = require('../auth/authHelper.js');
  */
 function executeRequestWithErrorHandling(req, res, next, requestType, requestPath, callback) {
 		if (req.cookies.REFRESH_TOKEN_CACHE_KEY === undefined) {
-			res.redirect(`/login?${process.env.host}&${process.env.ClientSecret}&${process.env.APPSETTING_ClientSecret}&${process.env.WEBSITE_HOSTNAME}`, next);
+			res.redirect(`/login?${process.env.BASE_URI}&${process.env.ClientSecret}&${process.env.APPSETTING_ClientSecret}&${process.env.WEBSITE_HOSTNAME}`, next);
```
Comment (jotrick) [inline Node/SampleApp/graph/httpsrequesthelper.js]: What are the Client Secret, APPSETTING_ClientSecret, and WEBSITE_HOSTNAME? (They weren't set as environment variables at the app.js level.  Also, why is one camel case, one both, and one caps locked?

File: Node/SampleApp/tabs/about.html [hunk 6]
```diff
@@ -30,13 +30,13 @@
...
-			<p>This is a sample app we made for you. This app will showcase all the awesome capabilities in Microsoft Teams</p>
-			<p>You can talk to a bot, get notifications, use compose extensions, add tabs and more</p>
+			<p>This is a sample app to showcase many of the awesome capabilities in Microsoft Teams</p>
+			<p>You can talk to a bot, use compose extensions, add tabs and more</p>
 			<p>The source code for this app is located at: <a href="https://aka.ms/teamssamples">aka.ms/teamsSamples</a></p>
 			<p>Video recordings of how this app was built are at: <a href="https://aka.ms/teamsbuildvideos">aka.ms/teamsBuildVideos</a> </p>
 			<p>Documentation on how to build apps for Microsoft Teams can be found at: <a href="https://aka.ms/teamsdocs">aka.ms/TeamsDocs</a></p>
 			<p>Submit your app here: <a href="https://aka.ms/submitteamsapp">aka.ms/SubmitTeamsApp</a></p>
-			<p>If you'd like to submit feedback, please use the feedback button at the bottom of the App Bar</p>
+			<p>If you'd like to submit feedback, please use the feedback button at the bottom of the App Bar.</p>
```
Comment (jotrick) [inline Node/SampleApp/tabs/about.html]: App Bar where?  in the aka.ms/SubmitTeamsApp?

File: Node/SampleNotificationBot/.vscode/launch.json [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+            }
+        },
+        {
+            "type": "node",
+            "request": "launch",
+            "name": "Launch Teams Debug",
+            "program": "${workspaceRoot}\\app.js",
+            "env": {
+                "NODE_ENV": "development",
+                "MICROSOFT_APP_ID": "[Your Bot ID Here]", //Bot ID from Bot Framework
```
Comment (jotrick) [inline Node/SampleNotificationBot/.vscode/launch.json]: Same comment as on the other launch.json as in, why are we filling them rather than leaving them blank?

File: Node/SampleNotificationBot/README.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
+# Teams Sample Notification bot for Node.js
+
+This sample shows a notification, or one-way bot.  This alternative to O365 Connectors utilizes the Bot Framework and all the rich functionality it provides to create a bot service that only pushes information into Microsoft Teams.  While it's easy to add basic conversational support, for some experiences a simple push system is all that is needed, and this will allow your notification service to full leverage the Bot APIs to target users and/or channels, and provide a rich experience utilizing Bot Framework cards.
```
Comment (jotrick) [inline Node/SampleNotificationBot/README.md]: "to fully leverage" = typo

File: Node/SampleNotificationBot/README.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        "scopes": [
+            "team",
+            "personal"
+        ]
+    }
+],
+...
+```
+
+This signals to Microsoft Teams that this is not a bot designed for bi-directional communication; rather, it is a bot designed to only send messages back to a team (`scope:teams`) or an individual user (`scope:personal`).  Notification Only bots should not be addressable in channels, and the personal context will only show "Activities", not "Conversations."
```
Comment (jotrick) [inline Node/SampleNotificationBot/README.md]: Should we add a note here that these features don't actually happen right now when the app is sideloaded (so they don't think there is a bug when they can at-mention the bot)?

File: Node/SampleNotificationBot/README.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+
+## Prerequisites
+
+The minimum prerequisites to run this sample are:
+* Latest Node.js with NPM. Download it from [here](https://nodejs.org/en/download/).
```
Comment (jotrick) [inline Node/SampleNotificationBot/README.md]: Is Node.js in the Prereqs for the other app?  Also, do we need to give them the instruction to call "npm install" to get the node modules?

File: Node/SampleNotificationBot/README.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+* **[Recommended]** Visual Studio Code for IntelliSense and debugging, download it from [here](https://code.visualstudio.com/) for free.
```
Comment (jotrick) [inline Node/SampleNotificationBot/README.md]: We only recommend VSCode here - should we remove Visual Studio from the other "Recommended" because it won't use the launch.json? 

Also, do we need to tell them how to run the launch.json with VSCode?

File: Node/SampleNotificationBot/README.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+* **[Optional]** The Bot Framework Emulator. To install the Bot Framework Emulator, download it from [here](https://emulator.botframework.com/). Please refer to [this documentation article](https://github.com/microsoft/botframework-emulator/wiki/Getting-Started) to know more about the Bot Framework Emulator.
+
+For more information about how to configure and test our samples, see [Sample applications for the Microsoft Teams Developer Platform](https://msdn.microsoft.com/en-us/microsoft-teams/samples).
+
+## Code Highlights
+
+This code simulates service-level notifications by create deep links to web endpoints used to send channel and direct user messages.  The deep links themselves contain  these endpoints (/send/user and /send/team) along with identifying logic like a GUID for lookup, and user and tenant IDs.
```
Comment (jotrick) [inline Node/SampleNotificationBot/README.md]: "This code simulates service-level notifications by create deep links to web endpoints used to send channel and direct user messages." First, typo = "creating".  Second, should we phrase this "deep links to web endpoints used to send channel and direct user messages" to something more like "deep links used to trigger the bot in order to get it to send channel and direct user messages"?  I just want it to be really clear that hitting these endpoints is triggering the bot to do something - not actually doing the sending or anything.

File: Node/SampleNotificationBot/README.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+The core logic is triggered by adding the bot to a channel, which sends the `conversationUpdate` event.  This event, when triggered by the bot addition itself, will create a Welcome message containing the deep links to the team channel and to individual team users.  Clicking on the links will launch the browser to those endpoints, and the Node server will process the event and send the appropriate message.
```
Comment (jotrick) [inline Node/SampleNotificationBot/README.md]: Should we not use "deep link" because they aren't links that "deep link" to a tab, but rather are just links?

File: Node/SampleNotificationBot/TeamsAppPackages/manifest.json [hunk 14]
```diff
@@ -1,22 +1,22 @@
 {
     "$schema": "https://statics.teams.microsoft.com/sdk/v1.0/manifest/MicrosoftTeams.schema.json",
     "manifestVersion": "1.0",
-    "version": "1.0",
-    "id": "9a1d84aa-225b-4856-83f8-ebaeca22b964",
-    "packageName": "com.microsoft.teams.samples.sampleapp",
+    "version": "1.0.0",
+    "id": "6c5a1f6a-b8da-4c75-93ca-e5e5cddd0999",
```
Comment (jotrick) [inline Node/SampleNotificationBot/TeamsAppPackages/manifest.json]: We could actually change this now to show what the go-forward plan is - have the bot.id and app.id be the same.  Also, should we have ids here or just tags like [registered_bot_id] so VSCode will underline it to yell about it when it doesn't validate to the schema (giving them a hint that they need to alter it)?

File: Node/SampleNotificationBot/TeamsAppPackages/manifest.json [hunk 15]
```diff
@@ -1,22 +1,22 @@
...
+    "packageName": "com.skype.teams.samples.sampleappnotify",
```
Comment (jotrick) [inline Node/SampleNotificationBot/TeamsAppPackages/manifest.json]: Why the change to Skype?  We are Microsoft Teams!!!

File: Node/SampleNotificationBot/app.js [hunk 16]
```diff
@@ -0,0 +1,299 @@
...
+                }
+            }, (err) => {
+
+            });
+
+        }
+    }
+});
+
+// Endpoint to send one way messages to the team
```
Comment (jotrick) [inline Node/SampleNotificationBot/app.js]: "Endpoint to trigger the  bot to send a one way message to the team"

File: Node/SampleNotificationBot/app.js [hunk 17]
```diff
@@ -0,0 +1,299 @@
...
+
+        bot.send(msg, function (err) {
+            // Return success/failure
+            res.status(err ? 500 : 200);
+            res.end();
+        });
+    } catch (e) { }
+});
+
+// Endpoint to send one way messages to individual users
```
Comment (jotrick) [inline Node/SampleNotificationBot/app.js]: "Endpoint to trigger the bot to send a one way message to an individual user"

File: Node/SampleApp/.vscode/launch.json (hunk 2 above)
Comment (richardtaylorrt) [inline Node/SampleApp/.vscode/launch.json]: @RichMoeMSFT @jotrick - let's chat quickly about how the sample app uses IDs. Guidance has changed as per our discussion on this, and the PR I just submitted for docs.

File: Node/SampleApp/README.md (hunk 3 above)
Comment (richardtaylorrt) [inline Node/SampleApp/README.md]: Yes, it would be great for the docs here - and the docs repo - to go deeper on auth.  I made the changes to the sample app to support auth, but never fully doc'd.  Happy to help you guys to do so now.
